app/bundles/CampaignBundle/Executioner/InactiveExecutioner.php

```diff
@@ -168,6 +171,19 @@ public function validate($decisionId, ContactLimiter $limiter, OutputInterface $
         return $this->counter;
     }
 
+    /**
+     * @internal Used in tests
+     */
+    public function setNowTime(\DateTime $dateTime): void
+    {
+        $this->now = $dateTime;
+    }
```

Comment:
Thanks for making this much needed optimization!

I'd like to prevent adding new code that is needed just for the tests. How about to extend this service and create special service just for test with the new methods needed to reset time? It can then be configured in config_test.php like https://github.com/mautic/mautic/blob/5.x/app/config/config_test.php#L143

Reply:
Created a separate class and decorated the service, so we'll not bother messing with class names and service ids in tests.